src/olympia/search/filters.py

```diff
@@ -189,6 +189,32 @@ def get_es_filter(self):
         return [Q('term', tags=tag) for tag in self.get_value()]
 
 
+class AddonFeaturedFilterParam(AddonFilterParam):
+    query_param = 'featured'
+    reverse_dict = {'': True, 'true': True}
+    valid_values = [True]
+    es_field = 'featured_for'
+
+    def get_es_filter(self):
+        self.get_value()  # Call to validate the value - we only want True.
+        app_filter = AddonAppFilterParam(self.request)
+        app = (app_filter.get_value()
+               if self.request.GET.get(app_filter.query_param) else None)
+        locale = self.request.GET.get('lang')
+        if not app and not locale:
+            # If neither app nor locale is specified fall back on is_featured.
+            return [Q('term', is_featured=True)]
+        queries = []
+        if app:
+            queries.append(
+                Q('term', **{'featured_for.application': app}))
+        if locale:
+            queries.append(
+                Q('terms', **{'featured_for.locales': [locale, 'ALL']}))
```

Comment:
comment suggestion:

```# Search for featured collections targeting `locale` or all locales.```

Somehow this got me confused during review often enough but I could be alone with this 😇